test/abstract-sql/get-referenced-fields.ts

```diff
@@ -0,0 +1,204 @@
+						['ReferencedField', 'table', 'field2'],
+					],
+				],
+				['From', ['Table', 'table']],
+			]),
+		).to.deep.equal({ table: ['field1', 'field2'] });
+	});
+
+	it('should work with filtered fields', () => {
+		expect(
+			AbstractSqlCompiler.postgres.getReferencedFields([
+				'SelectQuery',
+				['Select', []],
+				['From', ['Table', 'table']],
+				[
+					'Where',
+					[
+						'Equals',
+						['ReferencedField', 'table', 'field1'],
+						['ReferencedField', 'table', 'field2'],
+					],
+				],
+			]),
+		).to.deep.equal({ table: ['field1', 'field2'] });
+	});
+
+	it('should work with selected fields from an aliased table', () => {
+		expect(
+			AbstractSqlCompiler.postgres.getReferencedFields([
+				'SelectQuery',
+				[
+					'Select',
+					[
+						['ReferencedField', 'atable', 'field1'],
+						['ReferencedField', 'atable', 'field2'],
+					],
+				],
+				['From', ['Alias', ['Table', 'table'], 'atable']],
+			]),
+		).to.deep.equal({
+			table: ['field1', 'field2'],
+			atable: ['field1', 'field2'],
+		});
+	});
+
+	it('should work with filtered fields from an aliased table', () => {
+		expect(
+			AbstractSqlCompiler.postgres.getReferencedFields([
+				'SelectQuery',
+				['Select', []],
+				['From', ['Alias', ['Table', 'table'], 'atable']],
+				[
+					'Where',
+					[
+						'Equals',
+						['ReferencedField', 'atable', 'field1'],
+						['ReferencedField', 'atable', 'field2'],
+					],
+				],
+			]),
+		).to.deep.equal({
+			table: ['field1', 'field2'],
+			atable: ['field1', 'field2'],
+		});
+	});
+
+	it('should work with selected fields from multiple tables', () => {
+		expect(
+			AbstractSqlCompiler.postgres.getReferencedFields([
+				'SelectQuery',
+				[
+					'Select',
+					[
+						['ReferencedField', 'table', 'field1'],
+						['ReferencedField', 'table2', 'field1'],
+					],
+				],
+				['From', ['Table', 'table']],
+				['From', ['Table', 'table2']],
+			]),
+		).to.deep.equal({ table: ['field1'], table2: ['field1'] });
+	});
+
+	it('should work with filtered fields', () => {
+		expect(
+			AbstractSqlCompiler.postgres.getReferencedFields([
+				'SelectQuery',
+				['Select', []],
+				['From', ['Table', 'table']],
+				['From', ['Table', 'table2']],
+				[
+					'Where',
+					[
+						'Equals',
+						['ReferencedField', 'table', 'field1'],
+						['ReferencedField', 'table2', 'field1'],
+					],
+				],
+			]),
+		).to.deep.equal({ table: ['field1'], table2: ['field1'] });
+	});
+
+	it('should work with selected fields from multiple aliased tables', () => {
+		expect(
+			AbstractSqlCompiler.postgres.getReferencedFields([
+				'SelectQuery',
+				[
+					'Select',
+					[
+						['ReferencedField', 'atable', 'field1'],
+						['ReferencedField', 'atable2', 'field1'],
+					],
+				],
+				['From', ['Alias', ['Table', 'table'], 'atable']],
+				['From', ['Alias', ['Table', 'table2'], 'atable2']],
+			]),
+		).to.deep.equal({
+			table: ['field1'],
+			atable: ['field1'],
+			table2: ['field1'],
```

Comment:
I know I'm being somewhat pedantic here, but it would be better to assign different names to the fields since that would catch the case where the same list is accidentally referenced multiple times by the implementation.